Source/Gauntlet/Asserts/ThrowsAsserts.swift

```diff
@@ -106,16 +106,16 @@ public func XCTAssertThrowsError<T: Error>(
 ///   - then:       A closure to be called if the thrown `Error` is the expected type.
 ///                 This can be used to run additional tests with the error.
 @available(iOS 13.0.0, tvOS 13.0.0, macOS 10.15.0, *)
-public func XCTAssertThrowsError<T: Sendable, E: Error>(
+public func XCTAwaitAssertThrowsError<T: Sendable, E: Error>(
```

Comment:
To avoid making breaking changes, and needing to do a major version release, I think we could re-add the old function signature and have it call through to this newly named one. I think the name change is good for consistency and visibility, but would like to avoid a breaking change if possible. Those old signatures could also be marked as Deprecated so that they generate warnings when used and encourage people to use the new signature. 
That will mean the old names will hang around a bit, but we can remove those completely later when we do make a breaking change, and have a bigger reason to do it.

Reply:
Got it, makes sense. I guess re-adding the previous API and marking it as `@deprecated` may work, right?

Reply:
Yep, those changes look right. The only other thing I would add is one smoke check unit test for each of these that calls the deprecated function signature just to validate that API still works.

Reply:
@EmilioOjeda It doesn't look like these smoke check tests were added (or if they were I didn't seem them looking back). If we can get those added I can get this merged.

Reply:
@davidbarry-kr I have included test cases to verify that `XCTAssertThrowsError` calls `XCTAwaitAssertThrowsError`.

For the deprecated `XCTAssertNoThrow` async/await function, the compiler is not allowing me to add the test cases 🤔.

Reply:
Sounds good @EmilioOjeda , the changes look good. Looks like there are a few Swiftlint errors for some lines that are too long(The CI was previously not running, but is running now). This'll be good to merge once it passes lint.

Reply:
@davidbarry-kr  SwiftLint issues addressed.